dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/TasksRunner.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Stopwatch;
+import com.google.edwmigration.dumper.application.dumper.handle.Handle;
+import com.google.edwmigration.dumper.application.dumper.io.OutputHandle;
+import com.google.edwmigration.dumper.application.dumper.io.OutputHandleFactory;
+import com.google.edwmigration.dumper.application.dumper.task.Task;
+import com.google.edwmigration.dumper.application.dumper.task.TaskGroup;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import com.google.edwmigration.dumper.application.dumper.task.TaskSetState;
+import com.google.edwmigration.dumper.application.dumper.task.TaskSetState.Impl;
+import com.google.edwmigration.dumper.application.dumper.task.TaskState;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang3.time.DurationFormatUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** @author ishmum */
+public class TasksRunner {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TasksRunner.class);
+  private static final Logger PROGRESS_LOG = LoggerFactory.getLogger("progress-logger");
+
+  private AtomicInteger atomicNumberOfCompletedTasks;
+  private final int totalNumberOfTasks;
+  private final Stopwatch stopwatch;
+
+  private final TaskRunContext context;
+  private final TaskSetState.Impl state;
+  private final List<Task<?>> tasks;
+
+  public TasksRunner(
+      OutputHandleFactory sinkFactory,
+      Handle handle,
+      int threadPoolSize,
+      @Nonnull TaskSetState.Impl state,
+      List<Task<?>> tasks) {
+    context = createContext(sinkFactory, handle, threadPoolSize, state);
+    this.state = state;
+    this.tasks = tasks;
+    totalNumberOfTasks = countTasks(tasks);
+    stopwatch = Stopwatch.createStarted();
+    atomicNumberOfCompletedTasks = new AtomicInteger();
```

Comment:
I would suggest removing the "atomic" prefix from the `atomicNumberOfCompletedTasks` variable. It is already part of the type.